fedocal/__init__.py

```diff
@@ -687,11 +687,18 @@ def auth_login():
 @APP.route('/logout/')
 def auth_logout():
     """ Method to log out from the application. """
-    if not authenticated():
-        return flask.redirect(flask.url_for('index'))
-    FAS.logout()
-    flask.flash(gettext('You have been logged out'))
-    return flask.redirect(flask.url_for('index'))
+    next_url = flask.url_for('index')
+    if 'next' in flask.request.values:  # pragma: no cover
+        next_url = flask.request.values['next']
```

Comment:
I think you need to protect against a variant of "covert redirect" here and check that ``next_url`` is either 1) a relative url or 2) actually begins with ``SITE_ROOT`` so that people cannot get sneakily redirect to http://butt.holdings